pwa/src/components/subscribers/subscriberForm.tsx

```diff
@@ -57,57 +57,52 @@ export const SubscriberForm: React.FC<SubscriberFormProps> = ({ subscriberId, en
     handleSetTableNames();
   }, [API, subscriberId]);
 
-  const handleSetSubscriber = () => {
-    setShowSpinner(true);
+  React.useEffect(() => {
+    if (subscriberId) {
+
+      if (( !subscriber || !sources || !endpoints || !tableNames) && !showSpinner) setShowSpinner(true);
+      if (( subscriber && sources && endpoints && tableNames) && showSpinner) setShowSpinner(false);
+    } else {
+
+      if (( !sources || !endpoints || !tableNames) && !showSpinner) setShowSpinner(true);
+      if (( sources && endpoints && tableNames) && showSpinner) setShowSpinner(false);
+    }
+  }, [subscriber, sources, endpoints, tableNames]);
```

Comment:
This dependency array should also contain `subscriberId` (because when it changes, the logic within this effect changes).